### ISSUE TITLE
Compiler warnings: "warning: narrowing conversion of ..."

### ISSUE THREAD

**Issue by mtedaldi** [adafruit/Adafruit_SSD1305#23]
- Arduino board:  Mega2560 rev3

- Arduino IDE version (found in Arduino -> About Arduino menu):  1.8.13

- Description: when compiling the code I get compiler warnings (see code below). It's not an error but it seems unfortunate to get warnings from a library because it could prevent me catching warning that really mean something.

- Solution (sort of): adding two explicit (uint8_t) conversions in front of "p" and "page_offset" (see code below)

- Thoughts: This "Solution" is only a workaround. I can't fand any reason, why the variables (which are alle declared as uint8_t in the first place) would be converted to "int". The right approach would probably be to tackle this probably uneccesary conversion.

Code:
```
// i2c functions
#include <Wire.h>


// display hardware related functions used by GFX
#include <Adafruit_SSD1305.h>
#include <splash.h>

// graphics functions for the display
#include <Adafruit_GFX.h>
#include <Adafruit_GrayOLED.h>
#include <Adafruit_SPITFT.h>
#include <Adafruit_SPITFT_Macros.h>
#include <gfxfont.h>

void setup() {

  Wire.begin(); // start I2C0 as master

  // init display
  #define OLED_RESET 32
  Adafruit_SSD1305 display(128, 32, &Wire, OLED_RESET);

}

```

Compiler Output
```
...Arduino\libraries\Adafruit_SSD1305\Adafruit_SSD1305.cpp: In member function 'virtual void Adafruit_SSD1305::display()':
...\Arduino\libraries\Adafruit_SSD1305\Adafruit_SSD1305.cpp:310:47: warning: narrowing conversion of '((176 + ((int)p)) + ((int)((Adafruit_SSD1305*)this)->Adafruit_SSD1305::page_offset))' from 'int' to 'uint8_t {aka unsigned char}' inside { } [-Wnarrowing]
     uint8_t cmd[] = {SSD1305_SETPAGESTART + p + page_offset,
...\Arduino\libraries\Adafruit_SSD1305\Adafruit_SSD1305.cpp:311:27: warning: narrowing conversion of '(16 + ((((int)page_start) + ((int)((Adafruit_SSD1305*)this)->Adafruit_SSD1305::column_offset)) >> 4))' from 'int' to 'uint8_t {aka unsigned char}' inside { } [-Wnarrowing]
                      0x10 + ((page_start + column_offset) >> 4),
                      ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...Arduino\libraries\Adafruit_SSD1305\Adafruit_SSD1305.cpp:312:51: warning: narrowing conversion of '((((int)page_start) + ((int)((Adafruit_SSD1305*)this)->Adafruit_SSD1305::column_offset)) & 15)' from 'int' to 'uint8_t {aka unsigned char}' inside { } [-Wnarrowing]
                      (page_start + column_offset) & 0xF};
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
```

Solution (sort of):
```
    uint8_t cmd[] = {SSD1305_SETPAGESTART + (uint8_t)p + (uint8_t)page_offset,
                     0x10 + ((page_start + column_offset) >> 4),
                     (page_start + column_offset) & 0xF};
    oled_commandList(cmd, sizeof(cmd));
```

**Comment by ladyada** [comment]
hihi wanna try submitting a PR? :)

**Comment by mtedaldi** [comment]
> 
> 
> hihi wanna try submitting a PR? :)

My workaround didn't even fix it (I just did not get the warning on the socond compiler run :( ). The longer I look at this the less I seem to understand why the problem happens.

What I saw in Adafruit_SSD1305.h line 74 made me think that I had found the issue:
``` 
private:
  int8_t page_offset = 0;
  int8_t column_offset = 0;
```
but changing those to uint8_t didn't change anything (but I'd still suggest changing them to uint8_t because they can seem to only be applied a few positive constant values in the code).

The next step was a closer look at the literals (because they are cast as int by default). But casting them to byte did not solve the issue either :(

So I'm lost on this one and have no solution that works.
